### DISCUSSION TITLE
How can I activate a function from OnKeyUp?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#37435] [Help] [answered]
For example I have this here:

```
const Hydraulic = () => {
  function myFunction() {
    useEffect(() => {
      var input, filter, table, tr, td, i, txtValue;
      input = document.getElementById("myInput");
      filter = input.value.toUpperCase();
      table = document.getElementById("myTable");
      tr = table.getElementsByTagName("tr");
      for (i = 0; i < tr.length; i++) {
        td = tr[i].getElementsByTagName("td")[1];
        if (td) {
          txtValue = td.textContent || td.innerText;
          if (txtValue.toUpperCase().indexOf(filter) > -1) {
            tr[i].style.display = "";
          } else {
            tr[i].style.display = "none";
          }
        }
      }
    }, []);
  }
  return (
```
^^^^   This function is called in an input field to filter a table... like this below:

```
<input
            type="text"
            id="myInput"
            onKeyUp={myFunction()}
            placeholder="Search for Part Number.."
            title="Type in a name"
          />
```

I am getting NO ERRORS....  its just the table doesn't sort, like if the onkeyup function just never runs.

**Comment by Elindo586** [comment]
BTW... this is a page coming from just a regular html site that I am trying to convert into Nextjs..   the function works fine in just plain html..

**Reply by icyJoseph** [reply]
Passing callbacks in JSX, unlike in HTML, is done without invoking the function.

```tsx
<input
            type="text"
            id="myInput"
            onKeyUp={onKeyUpHandler}
            placeholder="Search for Part Number.."
            title="Type in a name"
          />
```

Now the question is, why is there a `useEffect` nested on that function? I think the useEffect is not needed, only the body of the function.  There seems like a bunch of things are wrong here, not invoking the function directly as you pass it, is a good start.

**Comment by Elindo586** [comment, marked as answer]
Thanks...

How do I mark your response as an answer?


Yes... your solution worked

Do some like this:
` onKeyUp={myFunction}`

instead of:
` onKeyUp={myFunction()}`

I was using useEffect because of an error I was getting from the ` input = document.getElementById("myInput");`   but I am not getting that error anymore and the code is working...


Thank you very much for your help...

**Comment by icyJoseph** [comment]
Replying here as main answer:

> Passing callbacks in JSX, unlike in HTML, is done without invoking the function.
> 
> ```tsx
> <input
>             type="text"
>             id="myInput"
>             onKeyUp={onKeyUpHandler}
>             placeholder="Search for Part Number.."
>             title="Type in a name"
>           />
> ```
> 
> Now the question is, why is there a `useEffect` nested on that function? I think the useEffect is not needed, only the body of the function. There seems like a bunch of things are wrong here, not invoking the function directly as you pass it, is a good start.
